Update dashboard guide with content from MSL do/dont guide (#9310)
DataDog/integrations-core: docs/developer/guidelines/dashboards.md

## Best Practices

### General

> 1. When creating a new dashboard, select the default dashboard type (internally called multisize layout.)

Reviewer: ```suggestion
1. When creating a new dashboard, select the default dashboard type (internally called multisize layout).
```

1. Research the metrics supported by the integration and consider grouping them in relevant categories.
Important metrics that are key to the performance and overview of the integration should be at the top.

> ### Rules of Thumb

Reviewer: ```suggestion
```

Reviewer: Deleting because rules of thumb are "general" best practices (which already has a heading). If you need to distinguish the points below from the items in "General," the heading should be more specific about how they differ. (Also avoiding using the very colloquial English phrase.)

1. Always include an “about” group for the integration containing a brief description and helpful links. Also include an “overview” group of a few of the most important metrics at the top of a dashboard. Use a group banner image for the “about” section if one is available. We store banner images [here](https://github.com/DataDog/web-ui/tree/preprod/public/static/images/integration_dashboard) and link to them with relative links in the dashboard. The “about” section should contain content, not data; the “overview” section should contain data. Avoid making the “about” section full-width.

> ![Elasticsearch dashboard](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/example_dashboard.png)

Reviewer: These images aren't working in my preview; I think direct links to `raw.githubusercontent.com` usually contain `blob/`right before the `master/` part of the URL, but it probably depends on how this gets published (which I don't know about).

Author: They won't render at these URLs until they're merged. The URL should work since the current guide uses an image from the same directory with this URL structure: `https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/ignite_dash.jpeg`

2. Use Group widgets to title and group sections, rather than note widgets as you might on a screenboard. Use partial width groups to display groups side-by-side. Most dashboards should display every widget within a group.

![Group titles](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/group_titles.png)

3. Timeseries widgets should be at least 4 columns wide in order not to appear squashed on smaller displays

![Squished timeseries](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/squished_timeseries.png)

> 4. Stream widgets should be at least 6 columns wide for readability (half the dashboard width.) Avoid placing full-width stream widgets in the middle of a dashboard as they’ll “trap” scroll events. It’s also useful to put stream widgets in a group by themselves so they can be collapsed. Add an event stream only if the service monitored by the dashboard is reporting events. Use `sources:service_name`.

Reviewer: ```suggestion
4. Stream widgets should be at least 6 columns wide (half the dashboard width) for readability. Avoid placing full-width stream widgets in the middle of a dashboard as they’ll “trap” scroll events. It’s also useful to put stream widgets in a group by themselves so they can be collapsed. Add an event stream only if the service monitored by the dashboard is reporting events. Use `sources:service_name`.
```

![Stream widgets](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/stream_widgets.png)

> 5. Which widgets best represent your data? Try using a mix of widget types and sizes. Explore visualizations and formatting options until you’re confident your dashboard is as clear as it can be. Sometimes a whole dashboard of timeseries is ok, but other times variety can improve things. There are different types of metric widgets. The most commonly used are [timeseries](https://docs.datadoghq.com/dashboards/widgets/timeseries/), [query values](https://docs.datadoghq.com/dashboards/widgets/query_value/), and [tables](https://docs.datadoghq.com/dashboards/widgets/table/). For more information on the available widget types, see the [list of supported dashboard widgets](https://docs.datadoghq.com/dashboards/widgets/).

Reviewer: ```suggestion
5. Which widgets best represent your data? Try using a mix of widget types and sizes. Explore visualizations and formatting options until you’re confident your dashboard is as clear as it can be. Sometimes a whole dashboard of timeseries is ok, but other times variety can improve things. The most commonly used metric widgets are [timeseries](https://docs.datadoghq.com/dashboards/widgets/timeseries/), [query values](https://docs.datadoghq.com/dashboards/widgets/query_value/), and [tables](https://docs.datadoghq.com/dashboards/widgets/table/). For more information on the available widget types, see the [list of supported dashboard widgets](https://docs.datadoghq.com/dashboards/widgets/).
```

![Variety of widgets](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/variety.png)

> 6. Add a logo to the dashboard header. The integration logo will automatically appear in the header if the icon exists here and the integration_id matches the icon name. That means it will only appear when the dashboard your working on is made into the official integration board.

Reviewer: ```suggestion
6. Add a logo to the dashboard header. The integration logo will automatically appear in the header if the icon exists [here](???) and the `integration_id` matches the icon name. That means it will only appear when the dashboard you're working on is made into the official integration board.
```
I think it looks like the word "here" is supposed to be linked? Or where is it referring to?

Author: Whoops, yes will fix by adding this link: https://druids.us1.prod.dog/logos/logo-gallery?variant=avatar

!!! tip
    Adding `*=scope` as a template variable is useful since users can access all their own tags.

### Copy

1. Prefer concise graph titles that start with the most important information. Avoid common phrases such as “number of”, and don’t include the integration title e.g. “Memcached Load”.

| Verbose title (bad) | Concise title (good) |
| - | - |
| Number of Kubernetes events per node | Events per node |
| Total number of pending tasks in [$node_name] | Pending tasks: [$node_name] |
| Number of read/write operations | Read/write operations |
| Rate of connections to server | Connections to server - rate |
| Memcached Load | Load |

2. Avoid repeating the group title or integration name in every widget in a group, especially if the widgets are query values with a custom unit of the same name. Note the word “shards” in each widget title in the group named “shards”.

![Name repetition](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/name_repetition.png)

3. Always alias formulas

4. Group titles should be title case. Widget titles should be sentence case.

> 5. If you’re showing a legend, make sure the aliases are easy to understand

Reviewer: ```suggestion
5. If you’re showing a legend, make sure the aliases are easy to understand.
```

6. Graph titles should summarize the queried metric. Do not indicate the unit in the graph title because unit types are displayed automatically from metadata. An exception to this is if the calculation of the query represents a different type of unit.

### QA

> 1. Always check a dashboard at 1280px wide and 2560px wide to see how it looks on a smaller laptop and a larger monitor. The most common screen widths for dashboards are 1920, 1680, 1440, 2560, and 1280px respectively. They make up more than half of all dashboard page views combined.

Reviewer: ```suggestion
1. Always check a dashboard at 1280px wide and 2560px wide to see how it looks on a smaller laptop and a larger monitor. The most common screen widths for dashboards are 1920, 1680, 1440, 2560, and 1280px, making up more than half of all dashboard page views combined.
```

![QA at different widths](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/qa_widths.png)

!!! tip
    Use [`TV Mode`](https://docs.datadoghq.com/dashboards/screenboards/#tv-mode) to ensure that the whole dashboard fits your screen.

### Visual Style

1. Format notes to make them fit their use case. Try the presets “caption”, “annotation”, or “header”, or pick your own combination of styles. Avoid using the smallest font size for notes that are long or include complex formatting, like bulleted lists or code blocks.

![Note styles](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/note_styles.png)

> 2. Use colors to highlight important relationships and to improve readability, not for style. If several groups are related, apply the same group header color to all of them. If you’ve applied a green header color to a group, try making its notes green as well. If two groups are related, but one is more important, try using the “vivid” color on the important group and the “light” color on the less important group. Be careful not to overuse color e.g. don’t make every group on a dashboard vivid blue. 

Reviewer 2: suggestion: also include staying away from greys/muted yellow-tones in notes, as they come off as less bold/eye-catching and users may be less inclined to read what they say

Author: Added

![Header color usage](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/header_color_usage.png)

> 3. Use legends when they make sense. Legends make it easy to read a graph without having to hover over each series or maximize the widget. Make sure you use aliases so the legend is easy to read. Automatic mode for legends is a great option that hides legends when space is tight and shows them when there’s room.

Reviewer 2: Aliases IMO are super important; I wonder if they should have their entire own bullet?

Author: There is a dedicated bullet for aliasing formulas: https://github.com/DataDog/integrations-core/pull/9310/files#diff-37e86720fb2437ffabf8d3f477ad076e0acdbd10ed0929a1c4c4e0dff0226f78R109

![Use legends](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/use_legends.png)

4. If you want users to compare two graphs side-by-side, make sure their x-axes align. If one graph is showing a legend and the other isn’t, the x-axes won’t align - make sure they either both show a legend or both do not.

![X axes alignment](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/x_axes_alignment.png)

5. For timeseries, base the display type on the type of metric

| Types of metric | Display type |
| - | - |
| Volume (e.g. Number of connections) | `area` |
| Counts (e.g. Number of errors) | `bars` |
| Multiple groups or default | `lines` |

### Examples

Elasticsearch

![Elasticsearch in high density mode](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/example_dashboard_high_density.png)

- [x] Attention-grabbing “about” section with a banner image, concise copy, useful links, and a good typography hierarchy
- [x] A brief, annotated “overview” section with the most important statistics, right at the top

> - [x] Simple graph titles. Group titles in title-case.

Reviewer: ```suggestion
- [x] Simple graph titles. Group titles in title-case
```
(all list items with ending punctuation or without, not a mix)
